src/datamancer/io.nim

```diff
@@ -607,6 +566,73 @@ proc parseCsvString*(csvData: string,
   result = readCsvTypedImpl(data, csvData.len, countNonEmptyLines(csvData),
                             sep, header, skipLines, toSkip, colNames)
 
+proc readCsvFromUrl(url: string,
+              sep: char = ',',
+              header: string = "",
+              skipLines = 0,
+              toSkip: set[char] = {},
+              colNames: seq[string] = @[]
+             ): DataFrame =
+  ## Reads a DF from a web URL (which must contain a CSV file)
+  var client = newHttpClient()
+  return parseCsvString(client.getContent(url), sep, header, skipLines, toSkip, colNames)
+
+proc readCsv*(fname: string,
+              sep: char = ',',
+              header: string = "",
+              skipLines = 0,
+              toSkip: set[char] = {},
+              colNames: seq[string] = @[],
+              skipInitialSpace = true,
+              quote = '"',
+             ): DataFrame =
+  ## Reads a DF from a CSV file or a web URL using the separator character `sep`.
+  ## 
+  ## `fname` can be a local filename or a web URL. If `fname` starts with
+  ## "http://" or "https://" the file contents will be read from the selected
+  ## web server. No catching is performed so if you plan to read from the same
```

Comment:
```suggestion
  ## web server. No caching is performed so if you plan to read from the same
```

Reply:
Note that we adding caching wouldn't really be hard. One could use `getTempDir` from the stdlib and write the contents to a file. Then upon downloading a file we'd first check if a file of the same name (i.e. from last part of the URL) exists in the temp dir.

However, for now maybe it's better to keep it simple (caching then also opens up the question of "should we overwrite if some time has elapsed", adding a `force` option to always download, making sure we actually get the same directory from `getTempDir` { I think it always gives you a new one } etc.). So this is fine (and I was just thinking out loud).